### ISSUE TITLE
Nuget.exe restore help claims support for project.json which does not actually work

### ISSUE THREAD

**Issue by agr** [NuGet/Home#5202]
Nuget.exe Version: 4.3.0.0

nuget.exe claims it is able to restore the packages using project.json:
```
E:\dev\misc\112>nuget help restore
NuGet Version: 4.3.0.0
usage: NuGet restore [<solution> | <packages.config file> | <project.json> | <Microsoft Build project>] [options]
```

But when it is actually attempted it produces an error:
```
E:\dev\misc\112>nuget restore project.json
Invalid input 'project.json'. Provide the path of an msbuild solution file instead. Support for XProj and standalone project.json files has been removed, to continue working with legacy projects use NuGet 3.5.x from https://nuget.org/downloads
```

**Comment by mishra14** [comment]
@emgarten

**Comment by rrelyea** [comment]
@mishra14  - Please confirm the 4.3 milestone I just added

**Comment by mishra14** [comment]
@rrelyea this was merged on May 9, so 4.3 looks fine to me.
